File: frontend/src/brand.ts
// Single source of truth for the user-visible product name in the SPA.
//
// The name isn't settled — we may rebrand before launch. Every visible
// "Target Workspace" string goes through this constant so a future rename
// is a one-line change (or one env var at build time).
//
// Override via Vite env at build:
//
//   VITE_BRAND_NAME="Acme Targeting" npm run build
//
// Backend has its own mirror (`src/target_workspace/brand.py`) — keep
// them in sync at deploy time.

const fromEnv =
  typeof import.meta !== "undefined" &&
  import.meta.env &&
  typeof import.meta.env.VITE_BRAND_NAME === "string"
    ? import.meta.env.VITE_BRAND_NAME.trim()
    : "";

export const BRAND_NAME: string = fromEnv || "Target Workspace";

File: src/target_workspace/brand.py
"""Single source of truth for the product name.

The project name isn't settled — we may rebrand before launch. Every
user-visible string that hard-codes "Target Workspace" goes through this
constant so a future rename is a one-line change (or one env var).

Override at boot via `TW_BRAND_NAME`:

    TW_BRAND_NAME="Acme Targeting" uvicorn ...

The frontend has its own mirror (`frontend/src/brand.ts`) since the SPA
is built separately. They default to the same string; a deployer who
overrides the backend value should override the frontend value too
(via `VITE_BRAND_NAME` at build time).
"""

from __future__ import annotations

import os

BRAND_NAME: str = os.environ.get("TW_BRAND_NAME", "Target Workspace").strip() or "Target Workspace"
